cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -63,38 +57,66 @@ type templateData struct {
 	RecentVotes []vote
 }
 
-func main() {
-	var err error
+// app struct contains global state.
+type app struct {
+	// db is the global database connection pool.
+	db *sql.DB
+	// tmpl is the parsed HTML template.
+	tmpl *template.Template
+}
+
+// indexHandler handles requests to the / route.
+func (app *app) indexHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		if err := showTotals(w, r, app); err != nil {
+			log.Printf("showTotals: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	case "POST":
+		if err := saveVote(w, r, app); err != nil {
+			log.Printf("saveVote: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	default:
+		http.Error(w, fmt.Sprintf("HTTP Method %s Not Allowed", r.Method), http.StatusMethodNotAllowed)
+	}
+}
 
-	parsedTemplate, err = template.ParseFiles("templates/index.html")
+func main() {
+	parsedTemplate, err := template.ParseFiles("templates/index.html")
 	if err != nil {
 		log.Fatalf("unable to parse template file: %s", err)
 	}
 
+	app := &app{
+		tmpl: parsedTemplate,
+	}
+
 	// If the optional DB_TCP_HOST environment variable is set, it contains
 	// the IP address and port number of a TCP connection pool to be created,
-	// such as "127.0.0.1:3306". If DB_TCP_HOST is not set, a Unix socket
+	// such as "127.0.0.1:1433". If DB_TCP_HOST is not set, a Unix socket
```

Comment:
What does this port number come from? Wouldn't the default still be 3306?

Reply:
Yup. This was a mistake from copying that bit from another sample. Fixed.